examples/credentials/main.go

```diff
@@ -18,12 +18,13 @@ func main() {
 }
 
 func provider() p.Provider {
-	return infer.NewProvider().
-		WithResources(infer.Resource[*User, UserArgs, UserState]()).
-		WithModuleMap(map[tokens.ModuleName]tokens.ModuleName{
+	return infer.Wrap(p.Provider{}, infer.Options{
```

Comment:
It seems likely that there will be providers that are implemented entirely with `infer` and having to do `infer.Wrap(p.Provider{}, infer.Options{...})` isn't great usage.

Instead, could we have `infer.NewProvider(infer.Options{...})` or even `infer.Provider(infer.Options{...})`?

If we added that, would we still need `Wrap` to wrap an existing `p.Provider`?